vendor/openfx/OCIOFile.cpp

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenColorIO Project.
+
+#include "OCIOFile.h"
+#include "OCIOProcessor.h"
+#include "OCIOUtils.h"
+
+namespace OCIO = OCIO_NAMESPACE;
+
+namespace
+{
+
+    const std::string PLUGIN_TYPE = "OCIOFile";
+
+} // namespace
+
+OCIOFile::OCIOFile(OfxImageEffectHandle handle)
+    : ImageEffect(handle)
+    , dstClip_(0)
+    , srcClip_(0)
+    , srcCsNameParam_(0)
+    , inverseParam_(0)
+{
+    dstClip_ = fetchClip(kOfxImageEffectOutputClipName);
+    srcClip_ = fetchClip(kOfxImageEffectSimpleSourceClipName);
+
+    srcCsNameParam_ = fetchChoiceParam("src_cs");
+
+    inverseParam_ = fetchBooleanParam("inverse");
+
+    // Handle missing config values
+    restoreChoiceParamOption(*this, "src_cs", PLUGIN_TYPE);
+
+    fetchContextParams(*this, contextParams_);
+}
+
+void OCIOFile::render(const OFX::RenderArguments& args)
+{
+    // Get images
+    std::unique_ptr<OFX::Image> dst(dstClip_->fetchImage(args.time));
+    std::unique_ptr<OFX::Image> src(srcClip_->fetchImage(args.time));
+
+    // Get transform parameters
+    std::string srcCsName = getChoiceParamOption(srcCsNameParam_);
+  
+    bool inverse = inverseParam_->getValue();
+
+    // Create context with overrides
+    OCIO::ContextRcPtr context = createOCIOContext(contextParams_);
+
+    // Build transform
+    OCIO::FileTransformRcPtr tr = OCIO::FileTransform::Create();
+    tr->setSrc(srcCsName.c_str());
+
+    // Setup and apply processor
+    OCIOProcessor proc(*this);
+
+    proc.setDstImg(dst.get());
+    proc.setSrcImg(src.get());
+    proc.setRenderWindow(args.renderWindow);
+    proc.setTransform(context, tr, (inverse ? OCIO::TRANSFORM_DIR_INVERSE
+        : OCIO::TRANSFORM_DIR_FORWARD));
```

Comment:
Very minor nit pick ... what you've done here is not wrong, but it would be better to follow the indentation style of the other OFX plug-ins (which are more consistent with the OCIO coding style guidelines).